From 1590b664d88be8386a4664c2994b685187d1eb25 Mon Sep 17 00:00:00 2001
From: Damien Goutte-Gattat <dgouttegattat@incenp.org>
Date: Thu, 3 Aug 2017 22:56:49 +0200
Subject: [PATCH 1/6] gtk: Disable tooltips in keyboard-grabbing mode.

* gtk+-2:/pinentry-gtk-2.c (show_hide_button): Do not show the
tooltip if we attempt to grab the keyboard.
(create_window): Likewise.
--

For unclear reasons, those tooltips may interfere with grabbing
under some tiling window managers.

GnuPG-bug-id: 3297
Signed-off-by: Damien Goutte-Gattat <dgouttegattat@incenp.org>
---
 gtk+-2/pinentry-gtk-2.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/gtk+-2/pinentry-gtk-2.c b/gtk+-2/pinentry-gtk-2.c
index d467ec5..f17a702 100644
--- a/gtk+-2/pinentry-gtk-2.c
+++ b/gtk+-2/pinentry-gtk-2.c
@@ -516,7 +516,10 @@ show_hide_button_toggled (GtkWidget *widget, gpointer data)
     }
 
   gtk_label_set_markup (GTK_LABEL(label), text);
-  gtk_widget_set_tooltip_text (GTK_WIDGET(button), tooltip);
+  if (!pinentry->grab)
+    {
+      gtk_widget_set_tooltip_text (GTK_WIDGET(button), tooltip);
+    }
   g_free (tooltip);
 }
 
@@ -736,7 +739,7 @@ create_window (pinentry_t ctx)
 	  gtk_progress_bar_set_text (GTK_PROGRESS_BAR (qualitybar),
 				     QUALITYBAR_EMPTY_TEXT);
 	  gtk_progress_bar_set_fraction (GTK_PROGRESS_BAR (qualitybar), 0.0);
-          if (pinentry->quality_bar_tt)
+          if (pinentry->quality_bar_tt && !pinentry->grab)
 	    {
 #if !GTK_CHECK_VERSION (2, 12, 0)
 	      gtk_tooltips_set_tip (GTK_TOOLTIPS (tooltips), qualitybar,
-- 
2.13.6

